Bump deployment target to 16.0

iamgabrielma opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #11867 
<!-- Id number of the GitHub issue this PR addresses. -->

## Description:
This PR updates the minimum deployment target to be iOS 16. There's ~10 deprecation warnings regarding NavigationLinks that haven't been addressed since these will need additional work. Details below.

## Changes:
* Updated all targets to iOS 16
* Updated deprecated `Locale.regionCode?` to `Locale.region?.identifier`
* Addressed some lint warnings that -oddly- didn't show previously, regarding whitespace, or 160+chars usage per line.

## Not addressed:
A CUW P2 can be created for these:

**1. NavigationLink deprecation warnings:** 

We use these in the `LazyNavigationLink` component, and in several views:`InPersonPaymentsMenu`, , `PaymentMethodsView`, `PaymentsRow`, and `ShippingLabelPackageList`. 

The update generally requires to wrap the existing views in either a `NavigationStack` or a `NavigationSplitView`, which modifies the current navigation behaviour. In some cases (Payments Menu) it might be necessary to implement some sort of navigation routing for each navigation case. In order to be sure that the different rows and destinations work correctly, as well as the different IPP flows that rely on account state changes work as expected, we should do this separately from this PR and assure we're not introducing any bug.

**2. Some CoreTelephony properties been deprecated with no replacement:**
 
We use these in the "network information" fields used for Zendesk when a user sends a ticket, specifically:
`serviceSubscriberCellularProviders`, `carrierName`, `isoCountryCode`. From my research, these have been deprecated for privacy reasons.

## Testing instructions
- CI tests should pass
- Go to Menu > Site switcher > `+ Add a Store` > `Create New Store` > Follow the creation flow. Observe how a predefined country is recommended during creation.

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

iamgabrielma committed:
Update targets min version

iamgabrielma committed:
update target for pods

dangermattic commented:
<!--
  1 Error: PR requires at least one label...
  0 Warnings
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">PR requires at least one label.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

iamgabrielma committed:
Update release notes

iamgabrielma committed:
update WooFoundation targets

iamgabrielma committed:
Update deprecated regionCode

The compiler is not happy with the type ambiguity derived from using .map(). Updated to use conditional binding instead.

iamgabrielma committed:
Address lint warnings, whitespace, 160+ lines

iamgabrielma commented on Experiments/Experiments.xcodeproj/project.pbxproj:
```diff
@@ -3,7 +3,7 @@
 	classes = {
 	};
-	objectVersion = 53;
+	objectVersion = 54;
```
This `objectVersion` was added automatically when updating the target, and seems to indicate the Xcode compatibility version, which would require a minimum of Xcode 12 to load the project.

bozidarsevo reviewed (approved):
Great! Now we can remove bunch of code that we had for 15 vs 16 :)
